nomenclature/config.py

```diff
@@ -0,0 +1,43 @@
+from pathlib import Path
+from typing import Dict
+from pydantic import BaseModel, validator
+
+import yaml
+
+
+class DataStructureConfig(BaseModel):
+    """A class for configuration of a DataStructureDefinition
+
+    Attributes
+    ----------
+    config : dict
+        Attributes for configuring the DataStructureDefinition
+
+    """
+
+    config: Dict[str, Dict] = {}
```

Comment:
I would make the dimensions explicit here: 
```suggestion
    region: Dict[str, Any] = {}
```
and add more as we go along. 
Or to completely satisfy my love for pydantic:
```python
    region: Optional[RegionCodeListConfig]
...
class RegionCodeListConfig(BaseModel):
    country: Optional[bool]
```
This is all the `RegionCodeListConfig` class would need to be for now. Again, can be extended for new features. 
The advantage is that you get type checking for the `config.yaml` file for free and don't have to do it "manually" like in `codelist.py` line 572, 581 and 582.